src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -2875,5 +3025,179 @@ static object G(C c, I<string?> x, I<object?> y)
                 //             x: x] // warn 2
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "x").WithArguments("I<string?>", "I<string>", "x", "int C.this[I<string> x, I<object?> y]").WithLocation(15, 16));
         }
+
+        [Fact]
+        public void Conversions_01()
+        {
+            var source =
+@"class A<T> { }
+class B<T> : A<T> { }
+class C
+{
+    static void F(B<object> x)
+    {
+        A<object?> y = x;
+    }
+    static void G(B<object?> x)
+    {
+        A<object> y = x;
+    }
+    static void H(B<object>? x)
+    {
+        A<object?> y = x;
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (7,24): warning CS8619: Nullability of reference types in value of type 'B<object>' doesn't match target type 'A<object?>'.
+                //         A<object?> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("B<object>", "A<object?>").WithLocation(7, 24),
+                // (11,23): warning CS8619: Nullability of reference types in value of type 'B<object?>' doesn't match target type 'A<object>'.
+                //         A<object> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("B<object?>", "A<object>").WithLocation(11, 23),
+                // (15,24): warning CS8619: Nullability of reference types in value of type 'B<object>' doesn't match target type 'A<object?>'.
+                //         A<object?> y = x;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("B<object>", "A<object?>").WithLocation(15, 24));
+        }
+
+        [Fact]
+        public void Conversions_02()
+        {
+            var source =
+@"interface IA<T> { }
+interface IB<T> : IA<T> { }
+class C
+{
+    static void F(IB<object> x)
+    {
+        IA<object?> y = x;
+        y = x;
+    }
+    static void G(IB<object?> x)
+    {
+        IA<object> y = x;
+        y = x;
+    }
+    static void H(IB<object>? x)
+    {
+        IA<object?> y = x;
```

Comment:
>IA<object?> y = x; [](start = 8, length = 18)

I would also expect a warning about possible null reference assignment on this and the next line.